appservice/src/createAppService/SiteRuntimeStep.ts

```diff
@@ -167,15 +167,19 @@ export class SiteRuntimeStep extends AzureWizardPromptStep<IAppServiceWizardCont
         ];
     }
 
-    private sortQuickPicksByRuntime(runtimeItems: IAzureQuickPickItem<ILinuxRuntimeStack>[], runtime: string): IAzureQuickPickItem<ILinuxRuntimeStack>[] {
+    private sortQuickPicksByRuntime(runtimeItems: IAzureQuickPickItem<ILinuxRuntimeStack>[], recomendedRuntimes: Map<LinuxRuntimes, number>): IAzureQuickPickItem<ILinuxRuntimeStack>[] {
```

Comment:
The map seems over-complex. What if it was just a sorted array of recommendedRuntimes. For example, this function could be implemented like this:
```typescript
private sortQuickPicksByRuntime(runtimeItems: IAzureQuickPickItem<ILinuxRuntimeStack>[], recommendedRuntimes: string[]): IAzureQuickPickItem<ILinuxRuntimeStack>[] {
    function getPriority(item: IAzureQuickPickItem<ILinuxRuntimeStack>): number {
        const index: number = recommendedRuntimes.findIndex(runtime => item.data.name.includes(runtime));
        return index === -1 ? recommendedRuntimes.length : index;
    }
    return runtimeItems.sort((a, b) => getPriority(a) - getPriority(b));
}
```

Reply:
Cool, using `array` is much better!